File: javacommons/security/javasrc/com/nokia/mj/impl/security/utils/TelUtils.java
/*
* Copyright (c) 2008 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description:
*
*/

package com.nokia.mj.impl.security.utils;

public final class TelUtils
{
    public static NetworkCodes getNetworkCodes()
    {
        return _getNetworkCodes();
    }

    public static class NetworkCodes
    {
        public String mcc = null;
        public String mnc = null;

        public NetworkCodes(String mcc, String mnc)
        {
            this.mcc = mcc;
            this.mnc = mnc;
        }
    }

    private static native NetworkCodes _getNetworkCodes();
}

File: javacommons/security/src.linux/telutils.cpp
/*
* Copyright (c) 2007-2007 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description:
*
*/


#include "telutils.h"

using namespace java::security;

TelUtils* TelUtils::createInstance()
{
    return new TelUtils();
}

void TelUtils::getNetworkCodes(std::string& mnc, std::string& mcc)
{
    //JELOG2(EUtils);
    mnc += "000";
    mcc += "000";
}

void TelUtils::getImei(std::string& imei)
{
    //JELOG2(EUtils);
    imei += "000000000000000";
}
